python/lsst/sims/catUtils/mixins/sncat.py

```diff
@@ -0,0 +1,302 @@
+        # id for galaxyObj catalogDBObj
+        # (email from Scott)
+        return self.column_by_name(self.refIdCol)
+
+    @property
+    def numobjs(self):
+        return len(self.column_by_name('id'))
+
+    @compound('snra', 'sndec', 'z', 'vra', 'vdec', 'vr')
+    def get_angularCoordinates(self):
+        '''
+        Obtain the coordinates and velocity of the SN from the host galaxy
+
+        Returns
+        -------
+        `np.ndarray` of coordinara, dec, z, vra, vdec, and vr
+
+        '''
+        hostra, hostdec, hostz = self.column_by_name('raJ2000'),\
+            self.column_by_name('decJ2000'),\
+            self.column_by_name('redshift')
+        snra, sndec, snz, snvra, snvdec, snvr = self.SNCoordinatesFromHost(
+            hostra, hostdec, hostz)
+
+        return ([snra, sndec, snz, snvra, sndec, snvr])
+
+    @compound('c', 'x1', 'x0', 't0')
+    def get_snparams(self):
+        hostz, hostid, hostmu = self.column_by_name('redshift'),\
+            self.column_by_name('snid'),\
+            self.column_by_name('cosmologicalDistanceModulus')
+
+        vals = self.SNparamDistfromHost(hostz, hostid, hostmu)
+
+        return (vals[:, 0], vals[:, 1], vals[:, 2], vals[:, 3])
+
+    def get_SNsed(self):
+        """
+        returns a list of SN seds in `lsst.sims.photUtils.Sed` observed within
+        the spatio-temporal range specified by obs_metadata
+
+        """
+        c, x1, x0, t0, _z, ra, dec = self.column_by_name('c'),\
+            self.column_by_name('x1'),\
+            self.column_by_name('x0'),\
+            self.column_by_name('t0'),\
+            self.column_by_name('redshift'),\
+            self.column_by_name('raJ2000'),\
+            self.column_by_name('decJ2000')
+
+        SNobject = SNObject()
+
+        sedlist = []
+        for i in range(self.numobjs):
+            SNobject.set(z=_z[i], c=c[i], x1=x1[i], t0=t0[i], x0=x0[i])
+            SNobject.setCoords(ra=ra[i], dec=dec[i])
+            SNobject.mwEBVfromMaps()
+            sed = SNobject.SNObjectSED(time=self.obs_metadata.mjd,
+                                       bandpass=self.lsstBandpassDict,
+                                       applyExitinction=True)
+            sedlist.append(sed)
+
+        return sedlist
+
+    def get_time(self):
+
+        return np.repeat(self.mjdobs, self.numobjs)
+
+    def get_band(self):
+        bandname = self.obs_metadata.bandpass
+        return np.repeat(bandname, self.numobjs)
+
+    @compound('flux', 'mag', 'flux_err', 'mag_err')
+    def get_snbrightness(self):
+
+        c, x1, x0, t0, _z, ra, dec = self.column_by_name('c'),\
+            self.column_by_name('x1'),\
+            self.column_by_name('x0'),\
+            self.column_by_name('t0'),\
+            self.column_by_name('redshift'),\
+            self.column_by_name('raJ2000'),\
+            self.column_by_name('decJ2000')
+
+        SNobject = SNObject()
+        bandname = self.obs_metadata.bandpass
```

Comment:
since it is possible that obs_metadata.bandpass is a list (at least, for now it is), you should probably have a check here that raises an error if obs_metadata.bandpass is a list (since this getter appears to be only able to return the flux and magnitude in a single bandpass)